### PR TITLE
[15.0][MIG] base_recurrence: Migration to 15.0

### PR DESCRIPTION
**Pull request by ilyasProgrammer**

### CODE REVIEW

**Comment by francesco-ooops** [comment]
@rousseldenis good migration? :)

**Review by rvalyi — APPROVED** [review]
thanks!

File: base_recurrence/models/recurrence_mixin.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        When added to the first day of the period,
+        it gives the first day of the next period.
+        """
+        recurring_func = "_recurring_rule_type_" + recurring_rule_type
+        if hasattr(self, recurring_func):
+            return getattr(self, recurring_func)(interval=interval)
+        else:
+            raise NotImplementedError()
+
+    # HELPER FUNCTIONS #
```
Comment (geomer198) [inline base_recurrence/models/recurrence_mixin.py:42]: I think need add `api.model` decorator for all HELPER FUNCTIONS witch start with `_recurring`.

**Review by geomer198 — COMMENTED** [review]
Please check review comment.

File: base_recurrence/models/recurrence_mixin.py (hunk 1 above)
Comment (rvalyi) [reply base_recurrence/models/recurrence_mixin.py:42]: that is true! I missed it in my review.

Comment (ilyasProgrammer) [reply base_recurrence/models/recurrence_mixin.py:42]: ok

**Comment by francesco-ooops** [comment]
@geomer198 @rousseldenis good for you?

**Review by rousseldenis — APPROVED** [review]
Sorry. On holidays :-) @ilyasProgrammer thanks for your interest in this module

**Review by geomer198 — APPROVED** [review]
LGTM!

**Review by rvalyi — APPROVED** [review]

**Pull request merged**